service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/spouts/kafka/job/TaskExecutor.java

```diff
@@ -0,0 +1,335 @@
+import eu.europeana.cloud.service.dps.DpsTask;
+import eu.europeana.cloud.service.dps.InputDataType;
+import eu.europeana.cloud.service.dps.OAIPMHHarvestingDetails;
+import eu.europeana.cloud.service.dps.PluginParameterKeys;
+import eu.europeana.cloud.service.dps.storm.NotificationTuple;
+import eu.europeana.cloud.service.dps.storm.StormTaskTuple;
+import eu.europeana.cloud.service.dps.storm.spouts.kafka.QueueFiller;
+import eu.europeana.cloud.service.dps.storm.spouts.kafka.QueueFillerForLatestRevisionJob;
+import eu.europeana.cloud.service.dps.storm.spouts.kafka.QueueFillerForSpecificRevisionJob;
+import eu.europeana.cloud.service.dps.storm.utils.CassandraTaskInfoDAO;
+import eu.europeana.cloud.service.dps.storm.utils.TaskStatusChecker;
+import eu.europeana.cloud.service.mcs.exception.MCSException;
+import org.apache.storm.spout.SpoutOutputCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.util.ConcurrentModificationException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.*;
+
+import static eu.europeana.cloud.service.dps.storm.AbstractDpsBolt.NOTIFICATION_STREAM_NAME;
+
+public class TaskExecutor implements Callable<Void> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TaskExecutor.class);
+    private static final int INTERNAL_THREADS_NUMBER = 10;
+    private static final int DEFAULT_RETRIES = 3;
+    private static final int SLEEP_TIME = 5000;
+    public static final int MAX_BATCH_SIZE = 100;
+
+    private TaskStatusChecker taskStatusChecker;
+
+    private SpoutOutputCollector collector;
+    private CassandraTaskInfoDAO cassandraTaskInfoDAO;
+    private ArrayBlockingQueue<StormTaskTuple> tuplesWithFileUrls;
+
+    private DataSetServiceClient dataSetServiceClient;
+    private RecordServiceClient recordServiceClient;
+    private FileServiceClient fileClient;
+
+    private String mcsClientURL;
+
+    private String stream;
+    private DpsTask dpsTask;
+
+    public TaskExecutor(SpoutOutputCollector collector, TaskStatusChecker taskStatusChecker, CassandraTaskInfoDAO cassandraTaskInfoDAO, ArrayBlockingQueue<StormTaskTuple> tuplesWithFileUrls, DataSetServiceClient dataSetServiceClient, RecordServiceClient recordServiceClient, FileServiceClient fileClient, String mcsClientURL, String stream, DpsTask dpsTask) {
+        this.collector = collector;
+        this.taskStatusChecker = taskStatusChecker;
+        this.cassandraTaskInfoDAO = cassandraTaskInfoDAO;
+        this.tuplesWithFileUrls = tuplesWithFileUrls;
+
+        this.dataSetServiceClient = dataSetServiceClient;
+        this.recordServiceClient = recordServiceClient;
+        this.fileClient = fileClient;
+
+        this.mcsClientURL = mcsClientURL;
+
+        this.stream = stream;
+        this.dpsTask = dpsTask;
+    }
+
+    @Override
+    public Void call() throws Exception {
```

Comment:
If execute method threw an exception we should catch it. I think we can do it in here!